pkg/api/router.go

```diff
@@ -55,5 +43,40 @@ func CreateMux(apiFunctions ScratchDataAPI) *chi.Mux {
 
 	r.Mount("/api", api)
 
+	router := chi.NewRouter()
+	router.Use(middleware.Logger)
+	router.Use(jwtauth.Verifier(apiFunctions.tokenAuth))
+
+	// TODO breadchris renable auth
+	//router.Use(apiFunctions.Authenticator(apiFunctions.tokenAuth))
```

Comment:
You've glossed over something pretty tricky. Maybe there's a way for this to be pluggable for a proprietery version.

- If a user is running this for the first time (the `go run` case), we shouldn't have auth. They should just be able to use the thing.
- The common case for a deployment would be for everyone to be part of the same organization. So any login (whether basic auth or google sign in) would assume that anyone has access to the same resources. one-to-many for teams and users.
- The proprietary case is for there to be multiple users and multiple teams. Each user only has access to the resources for the teams they belong to. many to many for teams and users. This is only relevant in a hosted version of Scratch Data.